stackit/services/mariadb/instance/resource.go

```diff
@@ -622,3 +627,25 @@ func (r *instanceResource) loadPlanId(ctx context.Context, diags *diag.Diagnosti
 	}
 	diags.AddError("Invalid plan_name", fmt.Sprintf("Couldn't find plan_name '%s' for version %s, available names are:%s", planName, version, availablePlanNames))
 }
+
+func (r *instanceResource) loadPlanNameAndVersion(ctx context.Context, diags *diag.Diagnostics, model *Model) {
```

Comment:
Instead of duplicating this for the data sources, we could also pass in the client as a function parameter (as it's the same client for both resource and data source)

Reply:
Or even fetch the offerings outside this function, and make it a local function that just searches for the respective plan name and version

Reply:
I would go with the first option